amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
Using `JUJU_TEST` prefix is best, as amulet is just meant to make test writing easier, but it's not the only way to write tests. I'd be okay with extra constraints, I could see cases where the testing infrastructure may want to set variables for deployment.

I guess if we knew the use case for this it'd be better to decide the best name/behavior for it in Amulet.